### ISSUE TITLE
RA build queue isn't user friendly. 

### ISSUE THREAD

**Issue by GrayHatter** [OpenRA/OpenRA#10166]
I can not count the number of games I've lost because I've queued what I thought was a generous number of units. Only to run out of them, have too much cash, and see my build queue empty, just in time to get steamrolled. 

Two options that would fix this: 

Either play a warning sound when the queue is empty (similar to silo's needed).

Or, allow for setting an auto train unit. I.e. if I set my auto train unit to artillery, when the queue is empty, it'll auto build arty. But when queue any unit, it'll auto cancel the arty, and build that unit. Then once empty again, it'll automagically start building arty again.
##

**Comment by GraionDilach** [comment]
Or just watch out your queues. You only have 6 afterall.

**Comment by penev92** [comment]
"Just make the game play itself already."

**Comment by jaZzKCS** [comment]
I have the feeling that this will again divide ppl into hardened blox of Opinionville.

If I understand it correctly, then you can set any unit on auto-queue but that would result in that production facility only producing that unit until the auto is stopped. Also true for any other unit from that same structure that is set to auto?

Because we want to make sure that you cannot auto-queue unit compositions, becasue then I really have to agree with penev here ("Hjelp! My game is playing itself" - a.k.a.: "Look Ma, no hands!")

TL;DR - It would be overkill if you could auto-queue more than one unit (compositions). Also keep in mind that this would make auto-queueing unit compositions instantly possible in TD since multiple construcion facilities give you multiple queues.

I have the feeling that this would have a far greater impact on TD than on RA in bigger games.

**Comment by jaZzKCS** [comment]
referring to: https://github.com/OpenRA/OpenRA/pull/10169

**Comment by Aaron-Lloyd** [comment]
The only time I feel auto-queuing or infinite queuing would ever actually be useful is maybe in the late part of a long TD match where you could potentially have 3 MCVs, 4 war factories, 4 helipads, and a couple of barracks and the number of queues becomes (a bit) less manageable. it would maybe make sense to be able to dedicate specific production facilities to certain units in that case with an autoqueue.

**Comment by RoosterDragon** [comment]
At the risk of stating the obvious: It seems like a more reasonable solution to "I didn't notice the build queue was empty" would be to improve the UI so empty build queues were more obvious to the player.

**Comment by r34ch** [comment]
@RoosterDragon +1

I was going to do a PR which either slowly filled the appropriate queue icon with the yellow color - or at the very least have it change color to yellow to show 'in production' and change the ready/blocked color to something like red and green (depending on the complexity)

In fact I might dibs that for xmas when I have some time to myself.

**Comment by atlimit8** [comment]
The tabs are sprites, so either sprite frames would be needed for every state or some kind of overlay(s) would have to be applied.
Personally, I would like to see the number of items queued appear over the tab. Then either the text color could change to reflect current production status or use a small overlay icon for status.

**Comment by penev92** [comment]
Personally, I think this is a lot of wasted effort to solve a non-existing problem.
(That's the short version. If anyone is interested in the long version, they can look up "whining about production queues" in the IRC logs.)

**Comment by leodew** [comment]
@RoosterDragon +1

**Comment by GrayHatter** [comment]
What about adding the sweep time remaining overlay that shows over the building unit/structure?

E.g. if you're building a Ore Refinery, and it's half way there, the left half of the Structures tab would be highlighted?

That's not the game playing itself, It's just giving you a quick glance to see if your build queue is done, or still working.

**Comment by pchote** [comment]
#15021 introduced an infinite build queue concept for custom maps and mods to use.
